src/Connections/GenericHook.ts

```diff
@@ -292,8 +301,30 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         // TODO: Transform Slackdown into markdown.
         return msg;
     }
+    
+    public async loadMatrixScript(scriptPath: string) {
+        if (typeof scriptPath !== "string") {
+            throw Error('loadMatrixScript takes a string')
+        }
+        const matrixUri = MatrixUriEventRegex.exec(scriptPath);
+        if (!matrixUri) {
+            throw Error('Not a valid matrix path. Use the event URI scheme from https://spec.matrix.org/v1.3/appendices/#matrix-uri-scheme');
+        }
+        const [ _, type, prefixlessRoomId, eventId ] = matrixUri;
+        let roomId = "!" + prefixlessRoomId;
+        if (type === "r") {
+            // RoomAlias -> resolve
+            roomId = await this.as.botClient.resolveRoom("#" + prefixlessRoomId);
+        }
+        const eventData = (await this.as.botClient.getEvent(roomId, eventId)).content as GenericHookConnectionState;
```

Comment:
I think `eventData` can be `undefined`, which will cause the next line to crash.